pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -279,3 +284,50 @@ func (p *criStatsProvider) makeContainerStats(
 
 	return result
 }
+
+// removeTerminatedContainer returns the specified container but with
+// the stats of the terminated containers removed.
+func removeTerminatedContainer(containers []*runtimeapi.Container, stats []*runtimeapi.ContainerStats) []*runtimeapi.Container {
+	containerMap := make(map[containerID][]*runtimeapi.Container)
+	// Sort order by create time
+	sort.Slice(containers, func(i, j int) bool {
+		return containers[i].CreatedAt < containers[j].CreatedAt
+	})
+	for _, container := range containers {
+		// Filter out the container that isn't managed by a Pod
+		if container.PodSandboxId == "" {
+			continue
+		}
+		refID := containerID{
+			podRef:        buildPodRef(container.Labels),
+			containerName: kubetypes.GetContainerName(container.Labels),
+		}
+		containerMap[refID] = append(containerMap[refID], container)
+	}
+
+	// Create container stats map.
+	containerStatsMap := make(map[string]*runtimeapi.ContainerStats)
+	for _, stat := range stats {
+		containerStatsMap[stat.Attributes.Id] = stat
+	}
+
+	result := make([]*runtimeapi.Container, 0)
+	for _, refs := range containerMap {
+		if len(refs) == 1 {
+			result = append(result, refs[0])
+			continue
+		}
+		i := 0
+		for ; i < len(refs); i++ {
```

Comment:
I prefer check container state and only keep running containers, since we've already got container state.

Reply:
Done.

Reply:
It seems that this is not addressed.

Reply:
I have not got what you mean clearly,  done.